Documentation/migration.md

```diff
@@ -0,0 +1,81 @@
+# Migration from TPR to CRDs
```

Comment:
Maybe drop an introductory paragraph in here why we're doing this and that the Prometheus Operator will automatically perform this migration, and that typically a user will not have to interact.